kv/mdbx/kv_mdbx.go

```diff
@@ -579,6 +587,49 @@ func (tx *MdbxTx) ForPrefix(bucket string, prefix []byte, walker func(k, v []byt
 	}
 	return nil
 }
+
+func (tx *MdbxTx) Range(table string, fromPrefix, toPrefix []byte) (kv.Pairs, error) {
+	s, err := tx.newStreamCursor(table)
+	if err != nil {
+		return nil, err
+	}
+	s.toPrefix = toPrefix
+	s.nextK, s.nextV, s.nextErr = s.c.Seek(fromPrefix)
+	return s, nil
+}
+func (tx *MdbxTx) newStreamCursor(table string) (*cursor2stream, error) {
+	c, err := tx.Cursor(table)
+	if err != nil {
+		return nil, err
+	}
+	s := &cursor2stream{c: c, ctx: tx.ctx}
+	tx.streams = append(tx.streams, s)
+	return s, nil
+}
+
+type cursor2stream struct {
+	c            kv.Cursor
+	nextK, nextV []byte
+	nextErr      error
+	toPrefix     []byte
+	ctx          context.Context
+}
+
+func (s *cursor2stream) Close() { s.c.Close() }
+func (s *cursor2stream) HasNext() bool {
+	return s.toPrefix != nil && bytes.Compare(s.nextK, s.toPrefix) < 0
```

Comment:
if `s.toPrefix = nil`, then this will return false always? won't that make this not usable to iterate across entire table.


```
if s.toPrefix == nil {
   return s.nextK != nil
}
```
on top is needed i think

Reply:
oh then `tx.Range(Table, from, nil)` will automatically do the `tx.ForEach` job... probably it's great.

Reply:
fixed

Reply:
then `tx.ForPrefix(prefix)` is equal to `tx.Range(Table, prefix, dbutils.NextSubtree(prefix) )` 
seems `tx.Range` is somewhat useful...